sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -145,7 +145,8 @@ class StaxXmlParser(
   def doParseColumn(xml: String,
       parseMode: ParseMode,
       xsdSchema: Option[Schema]): Option[InternalRow] = {
-    val xmlRecord = UTF8String.fromString(xml)
+    // limit the size of the XML records to add in bad record exception
```

Comment:
Hm, shouldn't we have an option for this?

Reply:
Adding an option to fine tune corrupt record handling may be an overkill. 
`UTF8String.fromString` throws  `NegativeArraySizeException` for strings of size 1B. 
I can increase the limit to close to that (say 512 M) to avoid this exception. WDYT?

Alternatively, I can remove the limit and just do `lazy val` that would throw an the above exception when a large bad record is encountered.

Reply:
just `lazy val` sounds simpler ... or we can just add an internal `SQLConf` to control this. e.g.,

```scala
buildConf("spark.sql.analyzer.maxIterations")
    .internal()
    .doc("...")
    .version("4.0.0")
    ...
```

Reply:
Thx. Updated the PR with just `lazy val`